### DISCUSSION TITLE
Sensor power management

### DISCUSSION THREAD

**Discussion by ltawii** [zephyrproject-rtos/zephyr#32552] [Ideas] [answered]
Hello

Current sensor API does not low power management dynamically.
Existing driver implementation only supports Kconfig flags to configure the sensor at startup, but it is limiting depending on application needs.
Is there any work being planned to add the capability to provide API to provide low power interface to switch an external sensor to different low power modes dynamically ?

**Comment by henrikbrixandersen** [comment, marked as answer]
A sensor driver can implement support for the device power management infrastructure described here: https://docs.zephyrproject.org/latest/reference/power_management/index.html#device-power-management-infrastructure

An example of a sensor driver implementing this is the BME280 driver: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/bme280/bme280.c

Pull requests for adding device power management support to more drivers are very welcome.

**Reply by Snevzor** [reply]
Consider a sensor on a separate power rail because the sensor is not low power enough and we only want to enable its power before fetching a sample.
The power rail is controller by a GPIO.

How do we do this regardless of the power management states?
So even in `PM_DEVICE_STATE_ACTIVE` we want to turn of the sensor power rail. When enabling the sensor power rail we want the driver init functions to be called.

I'm sorry if this is not the right place to ask, but I found this discussion to be matching the most to what I'm looking for.
